bin/p2-exec/main.go

```diff
@@ -80,6 +92,27 @@ func main() {
 		}
 	}
 
+	if *podID != "" {
+		if *launchableName != "" {
+			log.Fatalf("Specified pod name %q, but no launchable name was specified", *podID)
+		}
+		if *cgroupName != "" {
+			log.Fatalf("Specified pod name %q, but no cgroupname name was specified", *podID)
+		}
+
+		if config := os.Getenv("CONFIG_PATH"); config != "" {
+			params := cgroupParams{
+				Type: PodCGroup,
+			}
+			err := cgEnter(config, *podID, *podID, params)
+			if err != nil {
+				log.Fatal(err)
+			}
+		} else {
+			log.Fatal("No CONFIG_PATH, cannot determine pod-level cgroup")
```

Comment:
It's worth noting that sometimes we tell people to use `p2-exec` to run things outside of a pod context, so this might not be set. maybe that's okay but it's redefining p2-exec as only a thing that should be scheduled as a runit service for a P2 deployed process

Reply:
Oh, I should have squashed all these commits into one I think; the two new commits are the rewritten version. Without the --pod-cgroup flag its behavior remains the same, so I'm not sure I actually understand how this redefines p2-exec?